ui-speedspacechart/src/components/layers/SpeedLimitTagsLayer.tsx

```diff
@@ -18,12 +18,44 @@ const SpeedLimitTagsLayer = ({ width, marginTop, store }: SpeedLimitTagsLayerPro
   const tooltip = useRef<tooltipInfos | null>();
 
   useEffect(() => {
-    const updateTooltip = async () => {
+    const updateCanvas = async () => {
       const currentCanvas = canvas.current as HTMLCanvasElement;
       const ctx = currentCanvas.getContext('2d') as CanvasRenderingContext2D;
-      tooltip.current = await drawSpeedLimitTags({ ctx, width, height: marginTop, store });
+      const restrictedStore = {
```

Comment:
Hm, what is the purpose of creating an object containing a subset of `store` here? It seems like `drawSpeedLimitTags()` just pulls in part of these props anyways.

Note, TypeScript is pretty liberal when it comes to type compatibility for objects. This compiles just fine:

```ts
const obj = { foo: 42, bar: 43 };
type Foo = { foo: number };
const asdf: Foo = obj;
```

Reply:
Good catch. I think the code behaved the exact same before I did this, but I got an eslint warning because useEffect only contained a subset of the store and to the linter it looked like I was using all of the store. I changed a few things around since I got this linting error, so I will check again.

Reply:
Yeah still needed if we want the lint to pass. We could also deactivate this warning, but I do not dare do something like that.

Reply:
What is the linting error exactly?

Reply:
calling await drawSpeedLimitTags({ctx, width, height: marginTop, store: store}) in the useEffect without an explicit store dependence (which we don't want, and neither do we want all of its attributes as dependences because that includes cursor) causes :

/home/alice/dev/osrd-ui/ui-speedspacechart/src/components/layers/SpeedLimitTagsLayer.tsx
  32:6  warning  React Hook useEffect has a missing dependency: 'store'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

Reply:
ACK. Could be simplified a tad with `pick()` from lodash, but I'm fine with the current logic.

Reply:
I wish.
Doing something like
`      const restrictedStore = {
        ...pick(store, ['speedLimitTags', 'ratioX', 'leftOffset', 'speeds']),
        layersDisplay: pick(store.layersDisplay, ['electricalProfiles', 'powerRestrictions']),
      };
`
still leads to the same linting issue, as the linter does not particularly care to know what pick does, it only sees that store is used somewhere in the function. I'm sure there is some way to sidestep this issue, perhaps with useMemo, but it's not immediatly obvious to me